core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1898,8 +1928,32 @@ struct mobj_ffa *thread_spmc_populate_mobj_from_rx(uint64_t cookie)
 	if (!mf)
 		goto out;
 
-	if (set_pages(descr->address_range_array,
-		      READ_ONCE(descr->address_range_count), num_pages, mf)) {
+	assert(frag_length <= length && frag_length <= my_rxtx.size);
+	offs += offsetof(struct ffa_mem_region, address_range_array);
+	range_count = (frag_length - offs) / sizeof(struct ffa_address_range);
```

Comment:
`frag_length < off` is also an error case, no?

Reply:
Agree, Done